source/content/modules-known-issues.md

```diff
@@ -291,6 +291,20 @@ $conf[‘schema_suppress_type_warnings’] = TRUE;
 
 ___
 
+## [Search Api Solr Date Sort](https://www.drupal.org/project/search_api_solr_date_sort)
+
+<ReviewDate date="2020-02-14" />
+
+**Issue**: The module overrides a class from the [Pantheon Apache Solr module](https://pantheon.io/docs/solr-drupal-7) that is responsible for a correct connection to the Pantheon's Apache Solr service. As a result a Solr connection is lost.
+
+**Solution**: Instead of patching the module, you can fix the issue with a custom module:
+
+1. Define a new class that inherits from the `PantheonApachesolrSearchApiSolrService` and contains logic from the `SearchApiSolrDateSortSolrService` (or vice a versa).
```

Comment:
Is defining a new class something done once you have a new custom module, or is it part of the creation of it? I'm not a Drupal person, so I don't know how much of this knowledge can be assumed.

When you say "contains logic", that's a bit too ambiguous for my taste. Are we copying logic directly from the module in question? Some of it, all of it? Can we provide specific examples in the form of files or excerpts of code?

Reply:
Hi, @alexfornuto. Thank you for the review.

It's my first contribution to the Pantheon's documentation. I've thought that it's not necessary to provide here a huge listings of code. A user must be familiar with Drupal to solve this module's issue (know how to create custom modules, how to implement hooks) on the Pantheon.

I can explain what I meant in this PR. After that I will be very thankful if you (or somebody) say me how to correct my PR to provide a better clarification for end user.

- create a custom module
- create a new .inc file (for example, includes/service.inc)
- connect the .inc file in the module's .info file via "files[]" construction
- here is a class SearchApiSolrDateSortSolrService - https://git.drupalcode.org/project/search_api_solr_date_sort/-/blob/7.x-1.x/includes/service.inc#L10
- here is a class PantheonApachesolrSearchApiSolrService - https://github.com/pantheon-systems/drops-7/blob/default/modules/pantheon/pantheon_apachesolr/Pantheon_Search_Api_Solr_Service.php#L101
- in our .inc file we need to define a class that will inherit both properties and methods from 2 above classes
- implement the hook_search_api_service_info_alter() in the .module file; it can be grabbed from the https://git.drupalcode.org/project/search_api_solr_date_sort/-/blob/7.x-1.x/search_api_solr_date_sort.module#L99 - just replace a class name by our class from the .inc file
- about Drupal module's weight - https://www.drupal.org/docs/7/creating-custom-modules/howtos/how-to-update-a-modules-weight
- we need to take care that our module has a greater weight than the "search_api_solr_date_sort" and "pantheon_apachesolr" modules - it guaranties that our implementation of the hook_search_api_service_info_alter() will be called last and our class from the .inc will be used to during a communication with Pantheon's Solr Sercive

So if describe that step by step with code examples it could lead to a standalone article instead of small section. That's why I decided to be short.

Reply:
You're totally right, that's too much information to add here. I think that if anything, this could be improved by sharing whichever of the links you listed here would be most helpful for the reader to reference.